Promtail build fails repeatedly

### Bug Description

CI fails to build promtail.

### To Reproduce

Trigger CI. https://github.com/canonical/loki-k8s-operator/actions/runs/3657177658/jobs/6180478910

### Environment

N/A

### Relevant log output

```shell
Run GOHOSTOS=linux GOARCH=amd64 make clients/cmd/promtail/promtail
CGO_ENABLED=1 go build -ldflags "-s -w -X github.com/grafana/loki/pkg/util/build.Branch=HEAD -X github.com/grafana/loki/pkg/util/build.Version=2.7.0 -X github.com/grafana/loki/pkg/util/build.Revision=1b627d8 -X github.com/grafana/loki/pkg/util/build.BuildUser=runner@fv-az301-52 -X github.com/grafana/loki/pkg/util/build.BuildDate=2022-12-09T12:06:06Z" -tags netgo -o clients/cmd/promtail/promtail ./clients/cmd/promtail
# github.com/grafana/regexp/syntax
Error: vendor/github.com/grafana/regexp/syntax/parse.go:1037:18: undefined: strings.Cut
# sigs.k8s.io/json/internal/golang/encoding/json
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:96:31: undefined: any
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:177:35: undefined: any
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:428:37: undefined: any
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:937:48: undefined: any
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:1143:45: undefined: any
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:1160:42: undefined: any
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:1195:52: undefined: any
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/decode.go:1261:42: undefined: any
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:158:16: undefined: any
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:175:22: undefined: any
Error: vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:175:22: too many errors
note: module requires Go 1.18
make: *** [Makefile:202: clients/cmd/promtail/promtail] Error 2
Error: Process completed with exit code 2.
```


### Additional context

_No response_

--- a/.github/workflows/build-promtail-release.yaml
+++ b/.github/workflows/build-promtail-release.yaml
@@ -34,7 +34,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.17.8
+          go-version: 1.18
       - name: Install systemd dependencies
         run: |
           sudo apt-get update
@@ -99,7 +99,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.17.8
+          go-version: 1.18
       - name: Build Promtail
         # We run the make-based build with CGO_ENABLED=0
         # because we want statically-linked binaries and

--- a/lib/charms/loki_k8s/v0/loki_push_api.py
+++ b/lib/charms/loki_k8s/v0/loki_push_api.py
@@ -766,7 +766,7 @@ class AlertRules:
                         # any string as a "wildcard" which the topology labels will
                         # filter down
                         alert_rule["expr"] = self.tool.inject_label_matchers(
-                            re.sub(r"%%juju_topology%%", r'job=".+"', alert_rule["expr"]),
+                            re.sub(r"%%juju_topology%%", r'job=~".+"', alert_rule["expr"]),
                             self.topology.label_matcher_dict,
                         )
 

--- a/tox.ini
+++ b/tox.ini
@@ -70,7 +70,7 @@ deps =
     lib: git+https://github.com/canonical/operator#egg=ops
 commands =
     charm: mypy {[vars]src_path} {posargs}
-    lib: mypy --python-version 3.5 {[vars]lib_path} {posargs}
+    lib: mypy --python-version 3.8 {[vars]lib_path} {posargs}
 
 [testenv:unit]
 description = Run unit tests

